content/src/content/docs/learn/tutorials/first-endpoint.mdx

```diff
@@ -0,0 +1,355 @@
+
+NodeRuntime.runMain(main)
+```
+
+And we can run the resulting code like so:
+
+```shell showLineNumbers=false
+bun index.ts
+```
+
+Our Express `/health` handler still works as expected, but we've wrapped the
+Express app in a way that will allow us to define endpoints using Effect
+while still responding to our existing endpoints in Express.
+
+There's a lot of new things happening at the bottom of our file. What is
+`Effect.gen`, what are those `yield*`s doing, what is `Effect.never`. For the
+time being we're going to set these questions aside. It looks intimidating now,
+but if you stick with this tutorial to the end, it will not only make sense, but
+start to become second nature.
+
+## Our new `/health` handler
+
+The Effect equivalent of our Express `/health` handler looks like this:
+
+```ts twoslash showLineNumbers=false
+import { Effect } from "effect"
+// ---cut---
+import { HttpServerResponse } from "@effect/platform"
+
+const health = Effect.sync(() => {
+  return HttpServerResponse.text("ok")
+})
+```
+
+`HttpServerResponse` is Effect's class for generating HTTP responses, we need to
+use it for any responses returned from Effect.
+
+The way we wire this into our Effect `router` is going to look a little bit
+strange, but please stay with me:
+
+```ts twoslash showLineNumbers=false
+import { Effect } from "effect"
+// ---cut---
+import { HttpServerResponse, HttpRouter } from "@effect/platform"
+
+const health = Effect.sync(() => {
+  return HttpServerResponse.text("ok")
+})
+
+const router = HttpRouter.empty.pipe(
+  HttpRouter.get("/health", health),
+)
+```
+
+What on earth are `Effect.sync` and `.pipe`?
+
+## The Effect Type
+
+At the core of Effect is... well, the `Effect`. You can think of an `Effect` as
+a lazy computation, similar to a function that hasn't been called yet.
+
+Here's an example of an `Effect` that returns the string `"Hello, world!"`:
+
+```ts twoslash showLineNumbers=false
+import { Effect } from "effect"
+// ---cut---
+Effect.sync(() => "Hello, world!")
+```
+
+<Aside type="note">
+Returning a static value like this isn't something you would do in practice.
+The examples below are intentionally simple string manipulations to demonstrate
+how to use the `Effect` type.
+</Aside>
+
+`Effect.sync` takes a function and creates an `Effect` out of it. If you need to
+do asynchronous work in an Effect, you can use `Effect.promise`:
+
+```ts twoslash showLineNumbers=false
+import { Effect } from "effect"
+// ---cut---
+Effect.promise(async () => "Hello, world!")
+```
+
+If we want to run these Effects, we can use `Effect.runSync` and
+`Effect.runPromise`:
+
+```ts twoslash showLineNumbers=false
+import { Effect } from "effect"
+// ---cut---
+const syncResult = Effect.runSync(
+  Effect.sync(() => "Hello, world!")
+)
+console.log(syncResult)
+// => "Hello, world!"
+
+const asyncResult = await Effect.runPromise(
+  Effect.promise(async () => "Hello, world!")
+)
+console.log(asyncResult)
+// => "Hello, world!"
+```
```

Comment:
Despite the commentary we received, I would advocate strongly **against** showing `Effect.runSync` at all. A user shouldn't care at all about function coloring. Any `Effect` can be run with `Effect.runPromise`, and we should probably just stick to this during the tutorial.